src/FirmwarePlugin/APM/ArduSubFirmwarePlugin.cc

```diff
@@ -287,6 +289,7 @@ const char* APMSubmarineFactGroup::_lightsLevel1FactName        = "lights 1";
 const char* APMSubmarineFactGroup::_lightsLevel2FactName        = "lights 2";
 const char* APMSubmarineFactGroup::_pilotGainFactName           = "pilot gain";
```

Comment:
So this is the first I'm noticing these. Although I think it will currently work I'd recommend against fact names with spaces in them. You could get hosed further down the road by some other change which can't handle spaces. Or if you fact name gets used in an unexpected ways like for example persistance or things like that. There aren't any places in the code where fact names have spaces.

Reply:
I just pushed another two commits taking spaces out of fact names